packages/workspace/src/workspace/nacl_files/nacl_file_update.ts

```diff
@@ -31,20 +33,75 @@ import {
   isTypeReference,
 } from '@salto-io/adapter-api'
 import { AdditionDiff, ActionName } from '@salto-io/dag'
-import { walkOnElement, WalkOnFunc, WALK_NEXT_STEP } from '@salto-io/adapter-utils'
-import { collections } from '@salto-io/lowerdash'
+import { inspectValue, getPath, walkOnElement, WalkOnFunc, WALK_NEXT_STEP } from '@salto-io/adapter-utils'
+import { collections, values } from '@salto-io/lowerdash'
 import { parser } from '@salto-io/parser'
 
 const { awu } = collections.asynciterable
+const { isDefined } = values
+
+const log = logger(module)
 
 // Declared again to prevent cyclic dependency
 const FILE_EXTENSION = '.nacl'
 
-export type DetailedChangeWithSource = DetailedChange & { location: parser.SourceRange }
+export type Location = parser.SourceRange & { indexInParent?: number }
+
+export type DetailedChangeWithSource = DetailedChange & {
+  location: Location
+  requiresIndent?: boolean
+}
 
 const createFileNameFromPath = (pathParts?: ReadonlyArray<string>): string =>
   `${path.join(...(pathParts ?? ['unsorted']))}${FILE_EXTENSION}`
 
+type PositionInParent = {
+  followingElementIDs: ElemID[]
+  indexInParent?: number
+}
+
+const getPositionInParent = (change: DetailedChange): PositionInParent => {
```

Comment:
nit: can we have a more specific type here?
IIUC this is only ever called with addition changes, so maybe we can define this as
```suggestion
const getPositionInParent = (change: DetailedChange & AdditionChange<unknown>): PositionInParent => {
```
or something like that?

Reply:
Done.